.github/workflows/npm-publish.yml

```diff
@@ -25,6 +25,9 @@ jobs:
         env:
           DOTNET_CLI_TELEMETRY_OPTOUT: true
         run: dotnet publish --configuration Release --self-contained --runtime linux-x64 Cmf.CustomerPortal.Sdk.sln
+      # Run Unit Tests
+      - name: Run Unit Tests
+        run: dotnet test Cmf.CustomerPortal.Sdk.sln --configuration Release --collect:"XPlat Code Coverage" --logger trx
```

Comment:
no build no restore

Reply:
And, actually, this should be done in the following order:
1. Restore
2. Build
3. Test
4. Publish